src/SW_netCDF_Input.c

```diff
@@ -0,0 +1,3404 @@
+                "The input column '%s' contains a value of 'NA' for the "
+                "NC variable name '%s'. This is a required column and "
+                "must have information.",
+                mustTestAttNames[attNum],
+                varName
+            );
+            return; /* Exit function prematurely due to error */
+        }
+    }
+
+    if (!isIndex) {
+        testVeg = (Bool) (key == eSW_InVeg &&
+                          (isLitter || isBio || isPctLive || isLAI));
+
+        if (key == eSW_InSoil) {
+            if (strcmp(inputInfo[varNum][SW_INZAXIS], "NA") == 0) {
+                LogError(
+                    LogInfo,
+                    LOGERROR,
+                    "The input column '%s' contains a value of 'NA' for the "
+                    "NC variable name '%s'. This is a required column and "
+                    "must have information.",
+                    "ncZAxisName",
+                    varName
+                );
+                return; /* Exit function prematurely due to error */
+            }
+        } else if (key == eSW_InWeather || key == eSW_InClimate || testVeg) {
+            if (strcmp(inputInfo[varNum][SW_INTAXIS], "NA") == 0) {
+                LogError(
+                    LogInfo,
+                    LOGERROR,
+                    "The input column '%s' contains a value of 'NA' for the "
+                    "NC variable name '%s'. This is a required column and "
+                    "must have information.",
+                    "ncTAxisName",
+                    varName
+                );
+                return; /* Exit function prematurely due to error */
+            }
+        }
+
+        if (key == eSW_InWeather) {
+            if (inWeathStrideInfo[SW_INSTRIDEYR] > -1) {
+                if (inWeathStrideInfo[SW_INSTRIDESTART] == -1 ||
+                    strcmp(inputInfo[varNum][SW_INSTPATRN], "NA") == 0) {
+
+                    LogError(
+                        LogInfo,
+                        LOGERROR,
+                        "The input column 'ncStrideStart' and/or "
+                        "'ncStridePattern' "
+                        "contains a value of 'NA' for the NC variable name "
+                        "'%s'. "
+                        "These are required columns when 'ncStrideYears' is "
+                        "not "
+                        "'Inf' and must have information.",
+                        varName
+                    );
+                    return; /* Exit function prematurely due to error */
+                }
+            }
+        }
+    }
+}
+
+/**
+@brief Given a certain input key, make sure that specified input columns
+are the same throughout each active input variable
+
+@param[in] inputInfo Attribute information for all input variables
+@param[in] readInVars Specifies which variables are to be read-in as input
+@param[out] LogInfo Holds information on warnings and errors
+*/
+static void check_inputkey_columns(
```

Comment:
`check_inputkey_columns()` is currently too strict for "ncVAxisName"

A non-NA value of ncVAxisName is required for SW2 variables with pattern “\<veg\>.*” (e.g., “\<veg\>.transp_coeff”, “\<veg\>.fCover”) but must be NA for all other SW2 variables even if part of the same inKey (e.g., “\<veg\>.fCover” has a ncVAxisName value while "bareGround.fCover" has NA)

Reply:
I currently have this change to reflect the following conditions:

* If the input variable is a general one (`<veg>.*`), we check to see that the V-axis is not 'NA'
    * If it is NA, fail
* Otherwise, if the input variable within the two keys soil and vegetation (the keys that hold the general variables) is not general, we make sure that the V-axis is NA
    * If the value is not NA, fail

I would like to confirm whether this is the preferred route to handle this column for these two input keys, or if there is a different preference when handling the V-axis for handling both general and non-general variables.

Reply:
The check needs to cover all inkeys: fail if either the input variable does not contain the pattern (\<veg\>.*) or the V-axis name is not 'NA' but not both or none (i.e., exclusive or); otherwise pass